### ISSUE TITLE
Statements without semicolon have wrong location info

### ISSUE THREAD

**Issue by gyandeeps** [eslint/espree#164]
Code: 

```js
function test(){
	switch(a) {
	    case "a": return console.log("A")
	    case "b": return console.log("B")
	}
}
```

`Case` node on line 3 has following location detail
```
loc.start.line = 3
loc.end.line = 4
```

Thats the issue. 
FYI, it works fine if line 3 and 4 has `;` at the end.

**Comment by nzakas** [comment]
I think this might be correct. ASI would work like this:

```js
function test(){
	switch(a) {
	    case "a": return console.log("A")
	    ;case "b": return console.log("B")
	;}
}
```

**Comment by gyandeeps** [comment]
its different then Esprima. try [here](http://esprima.org/demo/parse.html?code=function%20test()%7B%0Aswitch(a)%20%7B%0A%20%20%20%20case%20%22a%22%3A%20return%20console.log(%22A%22)%0A%20%20%20%20case%20%22b%22%3A%20return%20console.log(%22B%22)%0A%7D%0A%7D)

**Comment by lo1tuma** [comment]
Duplicate of #41.

Seems like esprima change this recently and now showing always the end location of the last token instead of the node.

**Comment by gyandeeps** [comment]
If I am not wrong, A lot of rules check for same node like this

```
if(loc.start.line === loc.end.line)
```

I think that would be a problem for those rules also. And since we always use `;` in our unit tests, maybe we have not faced anything like that. Something to think about.

**Comment by gyandeeps** [comment]
`Acorn` and `Esprima` both works same.

**Comment by lo1tuma** [comment]
> if(loc.start.line === loc.end.line)

This is only a problem if `loc.end` belongs to a node, it works fine with tokens.

I would prefer espree to be spec compliant, and keep the node loc and range information as it is.

**Comment by BYK** [comment]
We had to do https://github.com/eslint/eslint/pull/3515 to fix against this. You can see the sample code in https://github.com/eslint/eslint/issues/3514 or in the newly added test case.

**Comment by gyandeeps** [comment]
@nzakas something to check/lookout for since we are switching to Acorn.

**Comment by nzakas** [comment]
This should be fixed with Acorn

**Comment by nzakas** [comment]
Fixed in 3.0.0 via Acorn.
